src/DonationForms/DonationQuery.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Give\DonationForms;
+
+use Give\Framework\QueryBuilder\JoinQueryBuilder;
+use Give\Framework\QueryBuilder\QueryBuilder;
+
+/**
+ * An opinionated Query Builder for GiveWP donations and meta fields.
+ *
+ * @unreleased
+ *
+ * Example usage:
+ * (new DonationQuery)
+ *     ->form(1816)
+ *     ->between('2024-02-00', '2024-02-23')
+ *     ->sumIntendedAmount();
+ */
+class DonationQuery extends QueryBuilder
+{
+    /**
+     * @unreleased
+     */
+    public function __construct()
+    {
+        $this->from('posts', 'donation');
+    }
+
+    /**
+     * An opinionated join method for the donation meta table.
+     * @unreleased
+     */
+    public function joinMeta($key, $alias)
+    {
+        $this->join(function (JoinQueryBuilder $builder) use ($key, $alias) {
+            $builder
+                ->leftJoin('give_donationmeta', $alias)
+                ->on('donation.ID', $alias . '.donation_id')
+                ->andOn($alias . '.meta_key', $key, true);
+        });
+        return $this;
+    }
+
+    /**
+     * An opinionated where method for the donation form ID meta field.
+     * @unreleased
+     */
+    public function form($formId)
+    {
+        $this->joinMeta('_give_payment_form_id', 'formId');
+        $this->where('formId.meta_value', $formId);
+        return $this;
+    }
+
+    /**
+     * An opinionated whereBetween method for the completed date meta field.
+     * @unreleased
+     */
+    public function between($startDate, $endDate)
+    {
+        $this->joinMeta('_give_completed_date', 'completed');
+        $this->whereBetween('completed.meta_value', $startDate, $endDate);
```

Comment:
I didn't test it on my end, but I'm very surprised that this works since the `meta_value` type is `longtext` and not `datetime` :thinking:

Reply:
Tested this on my end and it works :sweat_smile: 
Interesting